src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java

```diff
@@ -138,40 +139,62 @@ public Stage createStage(UUID caseUUID, CreateStageRequest createStageRequest) {
 
         log.debug("Creating Stage: {} for case {}", createStageRequest, caseUUID);
 
-        // find current active stage to deactivate (assumes 1 or 0)
-        Optional<Stage> maybeActiveStage = stageRepository.findFirstByTeamUUIDIsNotNullAndCaseUUID(caseUUID);
-
-        UUID currentUserUUID = null;
-        if(maybeActiveStage.isPresent()) {
-            currentUserUUID = maybeActiveStage.get().getUserUUID();
+        Set<Stage> allStages = stageRepository.findAllByCaseUUIDAsStage(caseUUID);
+        Set<Stage> activeStages = allStages.stream().filter(stage -> stage.getTeamUUID() != null).collect(Collectors.toSet());
+        if (activeStages.size() > 1) {
+            log.warn("More than 1 active stage, all active stages will be deactivated");
         }
 
-        // Create new stage
-        Stage newStage = new Stage(caseUUID, createStageRequest.getType(),
-                null, null, createStageRequest.getTransitionNoteUUID());
+        Optional<Stage> existingStageOfRequestedType = allStages.stream().filter(stage -> stage.getStageType().equals(createStageRequest.getType())).findFirst();
+
+        final AtomicBoolean isRecreation = new AtomicBoolean(false);
+        final AtomicBoolean isRecreateOfCurrentlyActive = new AtomicBoolean(false);
+        final AtomicReference<Stage> stageToActivate = new AtomicReference<>(null);
+
+        existingStageOfRequestedType.ifPresentOrElse((Stage stage) -> {
+            log.debug("Stage type of {} exists for caseUUID: {}, recreating stage",createStageRequest.getType(), caseUUID);
+            stageToActivate.set(stage);
+            isRecreation.set(true);
+            isRecreateOfCurrentlyActive.set(activeStages.contains(stage));
+        },() -> {
+            log.debug("Stage type of {} does not currently exist for caseUUID: {}, creating new stage",createStageRequest.getType(), caseUUID);
+            stageToActivate.set(new Stage(caseUUID, createStageRequest.getType(),
+                    null, null, createStageRequest.getTransitionNoteUUID()));
+        });
 
         CaseData caseData = caseDataService.getCase(caseUUID);
-        calculateDeadlines(newStage, caseData);
+        calculateDeadlines(stageToActivate.get(), caseData);
+
+        stageRepository.save(stageToActivate.get());
+        log.info("Created Stage: {}, Type: {}, Case: {}, event: {}", stageToActivate.get().getUuid(), stageToActivate.get().getStageType(), stageToActivate.get().getCaseUUID(), value(EVENT, STAGE_CREATED));
```

Comment:
```suggestion
        log.info("Created Stage: {}, Type: {}, Case: {}", stageToActivate.get().getUuid(), stageToActivate.get().getStageType(), stageToActivate.get().getCaseUUID(), value(EVENT, STAGE_CREATED));
```

Don't need that bit of the message - its magic how logback sends it to Kibana